File: WebAPI/Controllers/CommuniqueTypeController.cs
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using ReachUp;

namespace ReachUpWebAPI.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class CommuniqueTypeController : ControllerBase
    {
        // OK 
        [Authorize(Roles = "adm")]
        [HttpGet("GetAll")]
        public async Task<IActionResult> GetAll()
        {
           return Ok(await new CommuniqueType().GetAll());
        }

        // OK 
        [Authorize(Roles = "adm")]
        [HttpPost]
        public async Task<IActionResult> Post([FromBody] CommuniqueType communiqueType)
        {
            if (communiqueType != null)
                return Ok(await communiqueType.Add());
            return BadRequest("Parameters are null");
        }

        // OK 
        [Authorize(Roles = "adm")]
        [HttpPatch]
        public async Task<IActionResult> Patch([FromBody] CommuniqueType communiqueType)
        {
            if (communiqueType != null)
                return Ok(await communiqueType.Update());
            return BadRequest("Parameters are null");
        }

        // OK 
        [Authorize(Roles = "adm")]
        [HttpDelete]
        public async Task<IActionResult> Delete(int id)
        {
            if (!string.IsNullOrWhiteSpace(id.ToString()))
                return Ok(await new CommuniqueType().Delete(id));
            return BadRequest("Parameters are null");
        }
    }
}

File: WebAdmin/src/Controllers/CommuniqueTypeController.js
class CommuniqueTypeController {
    api;
    constructor(){
        this.api = new ApiConnection();
    }

    async getAll()
    {
        return await
         this.api.httpAnonymousGet('CommuniqueType/GetAll');
    }

    async add(name)
    {
        return await
         this.api.httpAnonymousPost(
             'CommuniqueType/',
             {
                name: name
             }
        );
    }

    async update(id, name)
    {
        return await
         this.api.httpAnonymousPatch(
             'CommuniqueType/',
             {
                id: id,
                name: name
             }
        );
    }

    async delete(id)
    {
        return await
         this.api.httpAnonymousDelete(`CommuniqueType?id=${id}`);
    }
}